actionpack/lib/action_dispatch/middleware/server_timing.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require "active_support/notifications"
+
+module ActionDispatch
+  class ServerTiming
+    SERVER_TIMING_HEADER = "Server-Timing"
+
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      events = []
+      subscriber = ActiveSupport::Notifications.subscribe(/.*/) do |*args|
+        events << ActiveSupport::Notifications::Event.new(*args)
+      end
+
+      status, headers, body = @app.call(env)
+      ActiveSupport::Notifications.unsubscribe(subscriber)
```

Comment:
Should we unsubscribe in an `ensure` block, to ensure we don't leak subscriptions if the `@app` crashes? 🤔

Reply:
@gsamokovarov are you thinking about something along these lines?
```rb
status, headers, body = begin
  @app.call(env)
ensure
  ActiveSupport::Notifications.unsubscribe(subscriber)
end
```

Reply:
Yes, something like this should work.

Reply:
Done in b15a815417c05b8c564b71f499bb4a5c1d24beef.